blockchain/event_filter.go

```diff
@@ -102,10 +103,7 @@ func (e *EventFilter) Events(cToken *ContinuationToken, chunkSize uint64) ([]*Fi
 		}
 	}
 
-	filterKeysMap := make(map[felt.Felt]bool, len(e.keys))
-	for _, key := range e.keys {
-		filterKeysMap[*key] = true
-	}
+	filterKeysMaps := makeKeysMap(e.keys)
 
 	curBlock := e.fromBlock
 	// skip the blocks that we previously processed for this request
```

Comment:
I couldn't select line 116 for some reason.

I am just curious why the `if` statement on line 117 (and 131) is not `curBlock <= latest` instead of `curBlock != latest + 1`? Is it not possible that the user can send `e.toBlock`and `e.FromBlock` which is larger than our chain height?

Reply:
Yes but we update them to be `latest+1` at max

https://github.com/NethermindEth/juno/pull/806/files#diff-299bd13c9d85e40bab3e1c50f08f697257ed323a0db67685f628d71e3c635518L94-L103

Reply:
That is only on `e.toBlock`, there are no checks on `e.fromBlock` to be much larger than the latest.

Also, I think a check should ensure that `from !>= to`.